python/ray/experimental/compiled_dag_ref.py

```diff
@@ -33,18 +33,14 @@ def __init__(
         self._execution_index = execution_index
         # Whether ray.get() was called on this CompiledDAGRef.
         self._ray_get_called = False
-        self._reader_refs = (
```

Comment:
Only the shared memory channel has `_reader_ref`, so it will have issues when the channel is `CompositeChannel`. We should not have channel-specific logic in this file. If we include channel-specific logic and import some channels, it is highly likely to cause a circular import. Instead, a better approach is to print the channels, and the channels should define what is useful for debugging by overwriting `__repr__`.

Reply:
I think `__repr__` is intended to generate the "official" string representation of the object, therefore should be more formal and contain all relevant fields. On the other hand, `__str__` is for an informal string representation, and if we were to pick one for an incomplete string representation, `__str__` is probably better. Can we use `__str__` across the board?

If we don't like using `__str__`, I'm also fine removing this for now to avoid adding too much work to this PR.

Reply:
Update https://github.com/ray-project/ray/pull/45950/commits/d446febd0f5cea70add8a8593cd5c79313738fe2